src/components/Messages/ConsoleInterceptor.js

```diff
@@ -0,0 +1,20 @@
+import { error } from "../../slices/MessagesSlice";
+import store from "../../store";
+
+// List of error messages we wish to intercept
+const interceptedConsoleMessages = ["Wrong network, please switch to mainnet"];
+
+// Intercepts an error sent to console and dispatches it to the message framework.
+var consoleInterceptor = function (message) {
+  if (interceptedConsoleMessages.includes(message)) {
+    store.dispatch(error(message));
+  }
+  console._error_old(message);
+};
+consoleInterceptor.isInterceptor = true;
+
+// Replaces the console.error function by our interceptor
+if (console.error.isInterceptor != true) {
+  console._error_old = console.error;
+  console.error = consoleInterceptor;
```

Comment:
Have you taken a look at all usages of console.error? Do we **NEED** to show all the errors to users?

Reply:
HI! That's why I put a filter on which messages are shown

Reply:
As a matter of fact it would be better to directly substitute calls to console.error by calls to the message framework. But since I do not know the project architecture I did not want to go that deep.

Reply:
Oh interesting. a cool patch. we'll definitely use the framework going forward where appropriate. :)